presto-main/src/main/java/com/facebook/presto/type/StatisticalQuantileDigest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.type;
+
+import com.facebook.presto.StatisticalDigest;
+import io.airlift.slice.Slice;
+import io.airlift.stats.QuantileDigest;
+
+public class StatisticalQuantileDigest
```

Comment:
I don't think we need these subtypes.  We can make `StatisticalDigest` a container type and restrict what types it takes in to TDigest and QDigest.

Reply:
How is this resolved?

Reply:
QuantileDigest and TDigest do not share a base class/interface, so we are not able to create a container type.

Reply:
I'm wondering, why do we decide to have `StatisticalQuantileDigest`, which is just a wrapper on `QuantileDigest`. Why not have `QuantileDigest` directly implement the interface? Only `merge` might need to check on the type rest would just work right? What did I miss?